contrib/packs/tests/test_action_download.py

```diff
@@ -81,6 +81,17 @@ def mock_is_dir_func(path):
     return original_is_dir_func(path)
 
 
+def mock_get_gitref(repo, ref):
+    """
+    Mock get_gitref function which return mocked object if ref passed is
+    PACK_INDEX['test']['version']
+    """
+    if ref == 'v%s' % PACK_INDEX['test']['version']:
+        return mock.MagicMock(hexsha=PACK_INDEX['test']['version'])
+    else:
+        return None
```

Comment:
What if code flow reaches this path?
With `None` as a returned value it'll error in the `pack_management.py` code.

Try it out and provide something better for this case.

Except of that, everything looks pretty good :+1:

Reply:
get_gitref is called few times during download. It do return None sometime if reference is not found. For this test case, it is called twice. First time ref passed is `0.4.0` and it return None, and second time `v0.4.0` is passed and gitref is returned.

Reply:
Imagine regression introduced in the code and https://github.com/StackStorm/st2/blob/219c9a263feaa87e9696a5a9a2ccde3b3e6423ad/st2common/st2common/util/pack_management.py#L373
got reverted to its previous state.

Taken reverted code, try to run the tests we now have to understand what I'm meaning.